Assets/Plugins/CountlySDK/Services/InitializationCountlyService.cs

```diff
@@ -52,21 +58,17 @@ internal async Task OnInitializationComplete()
                  */
                     await _locationService.SendIndependantLocationRequest();
                 }
-            }
-
-            if (!_configModel.EnableManualSessionHandling) {
+            } else if (!_configModel.EnableManualSessionHandling) {
                 //Start Session
                 await _sessionService.BeginSessionAsync();
             }
 
-            if (_configModel.EnableTestMode) {
+            if (_configModel.EnableTestMode || !_consentService.CheckConsent(Consents.Push) || _configModel.NotificationMode == TestMode.None) {
```

Comment:
This "if" should be reworked so that there is not "return" statement. That is bad code flow.